marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mapping/mappable-entity-lib.sjs

```diff
@@ -35,32 +35,46 @@ function addRelatedMappableEntities(entityName, targetEntityName, entityProperti
       let relatedEntityName = relatedEntityType.substring(relatedEntityType.lastIndexOf('/') + 1);
       let joinPropertyName = entityPropertyValue["joinPropertyName"] ? entityPropertyValue["joinPropertyName"] : entityPropertyValue.items["joinPropertyName"]
       if(relatedEntityName==targetEntityName){
-        const entityMappingId = entityName + ":" + targetEntityName + "." + joinPropertyName;
-        const mappingTitle = entityName + " (" + entityPropertyName + " "+ relatedEntityName + ")";
-        entityResponseObject.entityMappingId = entityMappingId;
-        entityResponseObject.mappingTitle = mappingTitle;
-        setRelatedEntityMappings(responseArray.find(response => response.entityType == targetEntityName), mappingTitle, entityMappingId);
+        const targetEntityResponse = responseArray.find(response => response.entityType == targetEntityName);
+        let relatedEntityMapping;
+        if(targetEntityResponse.relatedEntityMappings){
```

Comment:
This looks like it should be a separate function - the function is already 30 lines long with 6 arguments, which is pretty complex - VSCode's sonarlint plugin gives it a complexity score of 21 (the plugin defaults to a max score of 15, so this gets flagged). 

You may also want to combine some of the arguments into an object with a self-describing name, such as "entityContext".

Reply:
@rjrudin sonarlint sounds cool, I'll have to check it out.